### PR TITLE
Redirect and Remove sap modules

### PR DESCRIPTION
**Pull request by rainerleber**
##### SUMMARY
Remove sap relevant modules because of migration to community.sap_libs

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

sapcar_extract
hana_query
sap_task_list_execute

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by rainerleber** [comment]
@felixfontein i opened a new pr because of file structure changes

File: meta/runtime.yml [hunk 1]
```diff
@@ -952,7 +952,7 @@ plugin_routing:
           modules. This has never been supported or documented, and will stop working
           in community.general 9.0.0.
     database.saphana.hana_query:
-      redirect: community.general.hana_query
+      redirect: community.sap_libs.sap_hdbsql
```
Comment (felixfontein) [inline meta/runtime.yml:955]: Please do not touch these entries, instead add new ones for the actual modules. These entries are for the old long-names of the modules.

File: changelogs/fragments/5592-redirect-remove-sap-modules.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+  - sapcar_extract - migrated to community.sap_libs.sapcar_extract
+  - sap_task_list_execute - migrated to community.sap_libs.sap_task_list_execute
+  - hana_query - migrated to community.sap_libs.sap_hdbsql
```
Comment (felixfontein) [inline changelogs/fragments/5592-redirect-remove-sap-modules.yml:4]: Please look at https://github.com/felixfontein/community.general/blob/abecb17d0aefde99183c8444ca5b19264e85f8d1/changelogs/fragments/hetzner-migration-removal.yml and create something similar. You can cut out the 2.9/2.10 part, the collection now requires 2.11+.

File: changelogs/fragments/5592-redirect-remove-sap-modules.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+removed_features:
+  - |
+    All ``sap`` modules have been removed from this collection.
+    They have been migrated to the `community.sap_libs <https://galaxy.ansible.com/community/sap_libs>`_ collection.
```
Comment (felixfontein) [inline changelogs/fragments/5592-redirect-remove-sap-modules.yml:4]:
```suggestion
    They have been migrated to the `community.sap_libs <https://galaxy.ansible.com/community/sap_libs>`_ collection.
    Redirections have been provided.
```

File: changelogs/fragments/5592-redirect-remove-sap-modules.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    Following modules are affected:
+    - sapcar_extract
+    - sap_task_list_execute
+    - hana_query
```
Comment (felixfontein) [inline changelogs/fragments/5592-redirect-remove-sap-modules.yml:8]:
```suggestion
    - hana_query
breaking_changes:
- |
  If you are not using this collection as part of Ansible, but installed (and/or upgraded) community.general manually, you need to make sure to also install ``community.sap_libs`` if you are using any of the ``sapcar_extract``, ``sap_task_list_execute``, and ``hana_query` ` modules.
  Without that collection installed, the redirects for these modules do not work.
```

**Review by felixfontein — APPROVED** [review]

File: changelogs/fragments/5592-redirect-remove-sap-modules.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    All ``sap`` modules have been removed from this collection.
+    They have been migrated to the `community.sap_libs <https://galaxy.ansible.com/community/sap_libs>`_ collection.
+    Redirections have been provided.
+    Following modules are affected:
+    - sapcar_extract
+    - sap_task_list_execute
+    - hana_query
+breaking_changes:
+  - |
+    If you are not using this collection as part of Ansible, but installed (and/or upgraded) community.general manually, you need to make sure to also install ``community.sap_libs`` if you are using any of the ``sapcar_extract``, ``sap_task_list_execute``, and ``hana_query` ` modules.
```
Comment (felixfontein) [inline changelogs/fragments/5592-redirect-remove-sap-modules.yml:12]:
```suggestion
    If you are not using this collection as part of Ansible, but installed (and/or upgraded) community.general manually, you need to make sure to also install ``community.sap_libs`` if you are using any of the ``sapcar_extract``, ``sap_task_list_execute``, and ``hana_query`` modules.
```

File: meta/runtime.yml [hunk 6]
```diff
@@ -3389,6 +3391,10 @@ plugin_routing:
         warning_text: You are using an internal name to access the community.general.redfish_info
           modules. This has never been supported or documented, and will stop working
           in community.general 9.0.0.
+    sapcar_extract:
+      redirect: community.sap.sapcar_extract
```
Comment (felixfontein) [inline meta/runtime.yml:3395]:
```suggestion
      redirect: community.sap_libs.sapcar_extract
```

File: meta/runtime.yml [hunk 7]
```diff
@@ -3389,6 +3391,10 @@ plugin_routing:
...
+    sap_task_list_execute:
+      redirect: community.sap.sap_task_list_execute
```
Comment (felixfontein) [inline meta/runtime.yml:3397]:
```suggestion
      redirect: community.sap_libs.sap_task_list_execute
```

**Review by felixfontein — COMMENTED** [review]
Isn't `_libs` missing in both these places?

**Comment by rainerleber** [comment]
> Isn't `_libs` missing in both these places?

Yes it is thanks for the hint 😀

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@rainerleber thanks for taking care of this!

**Pull request merged**